src/pkcs15init/pkcs15-lib.c

```diff
@@ -3958,6 +3958,24 @@ sc_pkcs15init_verify_secret(struct sc_profile *profile, struct sc_pkcs15_card *p
 
 found:
 	if (pin_obj)   {
+		/*
+		* In this place, check if the PIN has been already validated and the card login
+		* state is still active.
+		*
+		* If pin cache is disabled or the reader is using pinpad, we can get here
+		* with no PIN data. This is ok as we can not asynchronously invoke the prompt
+		* (unless the pinpad is in use).
+		*/
+		r = sc_pkcs15_get_pin_info(p15card, pin_obj);
```

Comment:
Does it need to be done? AFAIK that code is used when there is no pin information available (I am not actually sure when and if this can even happen). In case of pkcs15 cards, the pin objects should be in place all the time and when the pkcs15 is emulated, the pin objects are also created for every driver I am aware of.